### PR TITLE
pkp/pkp-lib#924 Don't display empty sections in the browse block output

### PR DESCRIPTION
**Pull request by NateWr**

### CODE REVIEW

File: plugins/blocks/browse/block.tpl [hunk 1]
```diff
@@ -28,7 +28,7 @@
 				</li>
 			{/if}
 
-			{if $browseCategories}
+			{if $browseCategories->count}
```
Comment (beghelli) [inline plugins/blocks/browse/block.tpl]: is count an object property? $browseCategories is an array right?

Comment (NateWr) [inline plugins/blocks/browse/block.tpl]: It's a [DAO](https://github.com/pkp/omp/blob/master/plugins/blocks/browse/BrowseBlockPlugin.inc.php#L99-L113), not an array. Happy to hear of a better way to handle this.

Comment (beghelli) [inline plugins/blocks/browse/block.tpl]: Oh, ok. Have no problem using object factories in templates, I would just call it browseCategoryFactory, to be clear...

Comment (NateWr) [inline plugins/blocks/browse/block.tpl]: Good idea! I've updated those variable names and the docs explaining them to be more clear.

**Comment by NateWr** [comment]
It looks like the `CatalogSearchTest::testBombSearch` test failed. Not sure how this PR could be responsible so I think it's safe to merge.

**Comment by beghelli** [comment]
Can't see the Travis build, if it's just one build that failed (either MySQL or postgres) than its ok to merge. But I would give it a reload just to make sure.

File: plugins/blocks/browse/block.tpl [hunk 2]
```diff
@@ -28,11 +30,11 @@
 				</li>
 			{/if}
 
-			{if $browseCategories}
+			{if $browseCategoryFactory->count}
```
Comment (asmecher) [inline plugins/blocks/browse/block.tpl]: Use `->getCount()` rather than accessing the member variable directly.

**Comment by NateWr** [comment]
Good catch. Should be good go now...

**Comment by asmecher** [comment]
Merged, thanks!

**Pull request merged**